docs/citycloud-snippets.md

```diff
@@ -0,0 +1,130 @@
+OS_SUBNET_CIDR="10.10.0.0/24" # (Guessing) CIDR of network
+OS_FLOATINGIP_POOL="" # Name of the pool to get floating ips from
+```
+
+Set up the clusters into respective folders
+
+```bash
+pushd kubespray
+for CLUSTER in "${SERVICE_CLUSTER}" "${WORKLOAD_CLUSTERS[@]}"; do
+  mkdir -p "inventory/${CLUSTER}/group_vars"
+  # shellcheck disable=SC2016
+  sed -e "s@^cluster_name = .*@cluster_name = \"${CLUSTER}\"@" \
+      -e "s@^public_key_path = .*@public_key_path = \"${PUB_SSH_KEY_FILE}\"@" \
+      -e "s@^image = .*@image = \"${OS_IMAGE_NAME}\"@" \
+      -e "s@^ssh_user = .*@ssh_user = \"ubuntu\"@" \
+      -e "s@^number_of_k8s_masters = .*@number_of_k8s_masters = ${OS_NUM_MASTERS}@" \
+      -e "s@^number_of_k8s_nodes = .*@number_of_k8s_nodes = ${OS_NUM_WORKERS}@" \
+      -e "s@^number_of_k8s_nodes_no_floating_ip = .*@number_of_k8s_nodes_no_floating_ip = 0@" \
+      -e "s@^flavor_k8s_master = .*@flavor_k8s_master = \"${OS_MASTERS_FLAVOR}\"@" \
+      -e "s@^#\?flavor_k8s_node = .*@flavor_k8s_node = \"${OS_WORKER_FLAVOR}\"@" \
+      -e "s@^network_name = .*@network_name = \"${CLUSTER}-network\"@" \
+      -e "s@^external_net = .*@external_net = \"${OS_EXT_NET_ID}\"@" \
+      -e "s@^subnet_cidr = .*@subnet_cidr = \"${OS_SUBNET_CIDR}\"@" \
+      -e "s@^floatingip_pool = .*@floatingip_pool = \"${OS_FLOATINGIP_POOL}\"@" \
+  < "contrib/terraform/openstack/sample-inventory/cluster.tfvars" > "inventory/$CLUSTER/cluster.tfvars"
+done
+popd
+```
+
+Check the `cluster.tfvars` file and make sure your settings are what you want.
+If you want to be able to ssh into the machines you need to set the `k8s_allowed_remote_ips` variable to something like:
+
+```tfvars
+k8s_allowed_remote_ips = ["1.2.3.4/32"]
+```
+
+Now you're ready to deploy the infrastructure
+
+```bash
+for CLUSTER in "${SERVICE_CLUSTER}" "${WORKLOAD_CLUSTERS[@]}"; do
+  pushd "kubespray/inventory/${CLUSTER}"
+  terraform init ../../contrib/terraform/openstack
+  terraform apply \
+    -var-file cluster.tfvars \
+    -auto-approve \
+    -state="tfstate-${CLUSTER}.tfstate" \
+    ../../contrib/terraform/openstack
+  popd
+done
+```
+
+If all went well the infrastructure should now be up and running
+
+## Compliantkubernetes kubespray
+
+To set up kubernetes with compliantkubernetes-kubespray you can follow these steps.
+
+Export your cluster path and PGP fingerprint
+
+```bash
+export CK8S_CONFIG_PATH=~/.ck8s/my-new-cluster
+export CK8S_PGP_FP=<Your PGP fingerprint>
+```
+
+Initialize the configuration with.
+
+```bash
+for CLUSTER in "${SERVICE_CLUSTER}" "${WORKLOAD_CLUSTERS[@]}"; do
+  ./bin/ck8s-kubespray init "${CLUSTER}" citycloud ~/.ssh/id_rsa
+  ln -s "$(pwd)/kubespray/inventory/${CLUSTER}/tfstate-${CLUSTER}.tfstate" "${CK8S_CONFIG_PATH}/${CLUSTER}-config/" || true
+  cp "kubespray/contrib/terraform/openstack/hosts" "${CK8S_CONFIG_PATH}/${CLUSTER}-config/inventory.ini"
+  chmod +x "${CK8S_CONFIG_PATH}/${CLUSTER}-config/inventory.ini"
```

Comment:
I know what you mean here, but it feels weird to make `inventory.ini` executable. Making `hosts` executable feels more expected. Anyway, no need to fix this documentation, but please add an issue to `compliantkubernetes-kubespray` to handle both "static" inventory.ini and "dynamic" hosts.

Reply:
This is a requirement for kubespray. Since it couldn't execute this script without having permission to execute it

Reply:
I get that the script needs to be executable for kubespray. However, I would prefer that executable to be called `hosts` (without extension) and not `something.ini`. The `ini` part feels misleading.

Reply:
Right, yes that is a bit misleading. And yes, creating a new issue for this is probably the best for now.